pageserver/src/pgdatadir_mapping.rs

```diff
@@ -284,6 +284,14 @@ impl Timeline {
         if let Some(_nblocks) = self.get_cached_rel_size(&tag, version.get_lsn()) {
             return Ok(true);
         }
+        let buf = version.get(self, DBDIR_KEY, ctx).await?;
+        let dbdirs = match DbDirectory::des(&buf).context("deserialization failure") {
+            Ok(dir) => Ok(dir.dbdirs),
+            Err(e) => Err(PageReconstructError::from(e)),
+        }?;
+        if !dbdirs.contains_key(&(tag.spcnode, tag.dbnode)) {
+            return Ok(false);
+        }
         // fetch directory listing
         let key = rel_dir_to_key(tag.spcnode, tag.dbnode);
```

Comment:
You can move this line higher up to use it in your contains_key call.

Reply:
Sorry, but it is not possible. `dbdirs` hash map has key of type `(Oid, Oid)`
while `rel_dir_to_key` returns `Key`.